BrightFutures/AsyncType+Debug.swift

```diff
@@ -0,0 +1,48 @@
+//
+//  AsyncType+Debug.swift
+//  BrightFutures
+//
+//  Created by Oleksii on 23/09/2016.
+//  Copyright © 2016 Thomas Visser. All rights reserved.
+//
+
+import Result
+
+public protocol LoggerType {
```

Comment:
Can you give me the source on Swift 3 naming guidelines with this statement? Because the only thing that I could find was [here](https://swift.org/documentation/api-design-guidelines/) and it says that:

- Protocols that describe what something is should read as nouns (e.g. `Collection`).

- Protocols that describe a capability should be named using the suffixes able, ible, or ing (e.g. `Equatable`, `ProgressReporting`).

And I couldn't find anything about the `Protocol` in the end of the protocol name